velox/common/base/CompareFlags.h

```diff
@@ -34,113 +34,115 @@ struct CompareFlags {
-
-           - If if elements comapre results are true, then result is true.
-             ex: [1, 1] = [1, 1] is true.
-
-        3. Rows: Follows the same logic as array, with fields being elements.
-          - If any two fields compare result is false, result is false
-            ex: (null, 1) = (null, 2) is false.
-
-          - If result is not false and any two fields compare result is
-            indeterminate, then the result is indeterminate.
-            ex: (null, 1) = (null, 1) is indeterminate.
-
-            - If all fields comapre results are true, then result is true.
-            ex: (1, 1) = (1, 1) is indeterminate.
-
-        4. Maps:
-          - Keys are compared first, if keys are not equal values are not
-            checked even if they have nulls and result is false.
 
-         - If keys are the same for all maps, values are compared by applying
-           the array compare logic explained above observing each map as an
-           array of values.
-           ex: {1:null, 2:2} = {1:null, 2:3} is false.
-           ex: {1:null, 2:2} = {1:null, 2:2} is indeterminate.
-           ex: {1:1, 2:2} = {1:1, 2:2} is true.
-
-     ## When equalsOnly=false:
-      The compare either returns a value or throws a user error if a null is
-      encountered before result is determined, as explained below:
-
-        1. Primitive types and top level nulls:
-          - Comparing null with anything else throws (note that functions like
-            >, < do not pass nulls to compare since they are default nulls
-            functions).
-
-          - Comparing top level nulls also throws.
-
-        2. Arrays:
-          - Only elements up to index min(rhs.size(), lhs.size()) are compared.
-
-          - Elements are compared in order starting from index 0.
-
-          - If all elements in the range above are the same, then sizes are
-            compared.
-            ex: [1, null] > [1] -> result is true, and null is not read.
-
-          - If any two elements are different but not null, then result is
-            determined and remaining uncompared elements are ignored.
-            ex: [1, null] > [2, null] -> result is false and nulls are not
-            compared.
-
-          - If a null element is encountered before result is determined then
-            the compare throws a user exception.
-            ex: [1, null, null] > [1, 2]-> throws.
-
-        3. Rows:
-          - Fields are compared in order starting from index 0.
-
-          - If any two fields are different but not null, then result is
-            determined and remaining uncompared elements are ignored.
-            ex: (1, null) > (2, null) -> result is false and nulls not compared.
-
-          - If a null element is encountered before result is determined then
-            the compare throws a user exception.
-            ex: (1, null) > (1, 2) -> throws.
+    /// This is the default null handling mode, in this mode nulls are treated
```

Comment:
nit: Drop "This is" and "in this mode"

`The default null handling mode where nulls are treated...`